cirq/ops/gate_operation_test.py

```diff
@@ -25,9 +25,17 @@ def test_gate_operation_init():
     assert v.qubits == (q,)
 
 
+def test_invalid_gate_operation():
+    three_qubit_gate = cirq.ThreeQubitGate()
+    single_qubit = [cirq.GridQubit(0, 0)]
+    with pytest.raises(ValueError):
```

Comment:
Add `match="..."` clause to verify it's the expected type error.

(We've been really lax on this in general in the code base, and I'm trying to tighten up w.r.t. it)

Reply:
I've added an assertion to check that the message is that of a three qubit gate complaining about not getting three qubits. I've added similar checks to the corresponding method in the `parallel_gate_operation` version.